### ISSUE TITLE
Sidebar: onHide not triggered via programmatic close

### ISSUE THREAD

**Issue by saramcicchi** [primefaces/primeng#16507]
### Describe the bug

If you change the variable tied to the sidebar's `visible` input to true, `onShow` is triggered. However if you change it to false, `onHide` is not triggered.

See #13933, #14254, and #14450 for related issues.

### Environment

Standard angular app

### Reproducer

https://stackblitz.com/edit/nbkkb5-cfek47?file=src%2Fapp%2Fsidebar-basic-demo.ts

### Angular version

18.0.1

### PrimeNG version

17.18.11

### Build / Runtime

Angular CLI App

### Language

TypeScript

### Node version (for AoT issues node --version)

18.20.0

### Browser(s)

Firefox, Chrome

**Comment by bgoorden** [comment]
I encountered the same issue. In addition to the onHide not being triggered. The "visibleChange" is also not triggered via a programmatic close.

**Comment by owenstafford2** [comment]
Comment for PrimeNG issue #16507 (https://github.com/primefaces/primeng/issues/16507)

---

This appears to have regressed in the **Drawer** component (the successor to Sidebar) on **PrimeNG 21.1.8 / Angular 21**.

`onHide` is not emitted when `visible` is set to `false` programmatically; it only fires on an interactive close (X button / mask click). `onShow` still fires correctly on `true`, so the asymmetry described in this issue is back.

Root cause looks identical — the `visible` setter only acts on the truthy branch (`primeng/drawer`):

```js
set visible(value) {
    this._visible = value;
    if (this._visible && !this.modalVisible) {
        this.modalVisible = true;
    }
    // no else branch → nothing runs on a programmatic `visible = false`,
    // so hide()/onHide are never reached
}
```

`visibleChange` also doesn't emit on a programmatic hide, so there's currently no event that reliably signals a code-driven close.

**Expected:** setting `[visible]="false"` (or via `[(visible)]`) emits `onHide`, matching `onShow` on open.
**Actual:** no `onHide` (and no `visibleChange`) on programmatic close; only interactive closes emit.

Given this was fixed once for Sidebar (#16542, 17.18.12), it looks like the fix didn't carry through the Sidebar → Drawer rewrite. Happy to open a fresh issue against Drawer if you'd prefer to track it separately.
